ownCloudTests/Security/PasscodeTests.swift

```diff
@@ -65,4 +71,160 @@ class PasscodeTests: XCTestCase {
+
+        // Assure that the passcode is disabled
+        AppLockManager.shared.lockEnabled = false
+        EarlGrey.select(elementWithMatcher: grey_accessibilityID("settingsBarButtonItem")).perform(grey_tap())
+        EarlGrey.select(elementWithMatcher: grey_accessibilityID("passcodeSwitchIdentifier")).perform(grey_turnSwitchOn(true))
+
+        // Tap the number buttons first time
+        EarlGrey.select(elementWithMatcher: grey_accessibilityID("number1Button")).perform(grey_tap())
+        EarlGrey.select(elementWithMatcher: grey_accessibilityID("number1Button")).perform(grey_tap())
+        EarlGrey.select(elementWithMatcher: grey_accessibilityID("number1Button")).perform(grey_tap())
+        EarlGrey.select(elementWithMatcher: grey_accessibilityID("number1Button")).perform(grey_tap())
+
+        // Tap the number buttons second time & cancel
+        EarlGrey.select(elementWithMatcher: grey_accessibilityID("number1Button")).perform(grey_tap())
+        EarlGrey.select(elementWithMatcher: grey_accessibilityID("number1Button")).perform(grey_tap())
+        EarlGrey.select(elementWithMatcher: grey_text("Cancel")).perform(grey_tap())
+
+        // Asserts
+        EarlGrey.select(elementWithMatcher: grey_accessibilityID("passcodeSwitchIdentifier")).assert(grey_switchWithOnState(false))
+
+        //Reset status
+        EarlGrey.select(elementWithMatcher: grey_text("ownCloud")).perform(grey_tap())
+    }
+
+    /*
+     * PASSED if: Correct error when second typing is different than the first one
+     */
+    func testEnterDifferentPasscodes() {
+
+        // Assure that the passcode is disabled
+        AppLockManager.shared.lockEnabled = false
+        EarlGrey.select(elementWithMatcher: grey_accessibilityID("settingsBarButtonItem")).perform(grey_tap())
+        EarlGrey.select(elementWithMatcher: grey_accessibilityID("passcodeSwitchIdentifier")).perform(grey_turnSwitchOn(true))
+
+        // Tap the number buttons
+        EarlGrey.select(elementWithMatcher: grey_accessibilityID("number1Button")).perform(grey_tap())
+        EarlGrey.select(elementWithMatcher: grey_accessibilityID("number1Button")).perform(grey_tap())
+        EarlGrey.select(elementWithMatcher: grey_accessibilityID("number1Button")).perform(grey_tap())
+        EarlGrey.select(elementWithMatcher: grey_accessibilityID("number1Button")).perform(grey_tap())
+
+        // Tap the number buttons again
+        EarlGrey.select(elementWithMatcher: grey_accessibilityID("number1Button")).perform(grey_tap())
+        EarlGrey.select(elementWithMatcher: grey_accessibilityID("number1Button")).perform(grey_tap())
+        EarlGrey.select(elementWithMatcher: grey_accessibilityID("number1Button")).perform(grey_tap())
+        EarlGrey.select(elementWithMatcher: grey_accessibilityID("number3Button")).perform(grey_tap())
+
+        // Asserts
+        EarlGrey.select(elementWithMatcher: grey_text("The entered codes are different")).assert(grey_sufficientlyVisible())
```

Comment:
You should use the text from the translated file because maybe the text change or the device for testing is in other language

Reply:
👍